WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCoupon.swift

```diff
@@ -1,9 +1,55 @@
 import SwiftUI
 import Yosemite
 
+final class AddEditCouponHostingController: UIHostingController<AddEditCoupon> {
+
+    private let viewModel: AddEditCouponViewModel
+
+    init(viewModel: AddEditCouponViewModel) {
+        self.viewModel = viewModel
+        super.init(rootView: AddEditCoupon(viewModel))
+
+        rootView.dismissHandler = { [weak self] in
+            guard let self = self else { return }
+            if viewModel.hasChangesMade {
+                UIAlertController.presentDiscardChangesActionSheet(viewController: self) { [weak self] in
+                    self?.dismiss(animated: true)
+                }
+            } else {
+                self.dismiss(animated: true)
+            }
+        }
```

Comment:
I think we should probably dismiss the view instantly here since we had a discussion about the close button earlier with the Eagle team (p1649409321754619/1649383906.341789-slack-C6H8C3G23), and the discard alert should only be presented on swipe gesture to make sure that the user didn't accidentally close the modal.

Reply:
Thanks for giving me this context! I've fixed it here: https://github.com/woocommerce/woocommerce-ios/pull/6889/commits/a159a30df8fe55f99852337a903bd57d99663ad0